src/components/filters/location.js

```diff
@@ -0,0 +1,203 @@
+// @flow
+import React from 'react';
+import { fromJS } from 'immutable';
+import { Async } from 'react-select';
+import Select from 'react-select';
+
+import bbox from '@turf/bbox';
+import simplify from '@turf/simplify';
+import MapboxDraw from '@mapbox/mapbox-gl-draw';
+
+import { nominatimSearch } from '../../network/nominatim';
+import { mapboxAccessToken } from '../../config/constants';
+import { importChangesetMap } from '../../utils/cmap';
+
+export class LocationSelect extends React.PureComponent {
+  props: {
+    name: string,
+    display: string,
+    value: Object,
+    type: string,
+    placeholder: string,
+    options: Array<Object>,
+    onChange: (string, value: Object) => any
+  };
+  state = {
+    geometry: Object,
+    location: '',
+    queryType: 'q',
+    lastSearch: new Date()
+  };
+  queryTypeOptions = [
+    { value: 'q', label: 'Any' },
+    { value: 'city', label: 'City' },
+    { value: 'county', label: 'County' },
+    { value: 'state', label: 'State' },
+    { value: 'country', label: 'Country' }
+  ];
+  map = null;
+  draw = null;
+
+  updateMap(data) {
+    if (this.map.getSource('feature')) {
+      this.map.getSource('feature').setData(data);
+    } else {
+      this.map.addSource('feature', {
+        type: 'geojson',
+        data: {
+          type: 'Feature',
+          geometry: data
+        }
+      });
+    }
+    if (this.map.getLayer('geometry') === undefined) {
+      this.map.addLayer({
+        id: 'geometry',
+        type: 'fill',
+        source: 'feature',
+        paint: {
+          'fill-color': '#088',
+          'fill-opacity': 0.6
+        }
+      });
+    }
+    this.setState({ geometry: data });
+    this.props.onChange(
+      this.props.name,
+      fromJS([{ label: data, value: data }])
+    );
+    const geom_bbox = bbox(data);
+    this.map.fitBounds(
+      [geom_bbox.slice(0, 2), geom_bbox.slice(2)],
+      {padding: 20}
+    );
+  }
+
+  componentDidMount() {
+    importChangesetMap('getGL').then((getGL: any) => {
+      if (getGL) {
+        var mapboxgl = getGL();
+        mapboxgl.accessToken = mapboxAccessToken;
+        const map = new mapboxgl.Map({
+          container: 'geometry-map',
+          style: 'mapbox://styles/mapbox/light-v9'
+        });
+        this.map = map;
+        this.draw = new MapboxDraw({
+          displayControlsDefault: false,
+          controls: {
+            polygon: true
+          }
+        });
+        map.addControl(this.draw);
+
+        map.on('draw.create', this.drawingUpdate);
+        map.on('draw.modechange', this.clearBeforeDraw);
+        map.on('draw.delete', this.drawingUpdate);
+        map.on('draw.update', this.drawingUpdate);
+        map.on('style.load', () => {
+          if (this.props.value.get('0').get('value')) {
+            this.updateMap(this.props.value.get('0').get('value').toJS());
+          }
```

Comment:
👋  @willemarcel, I am getting `TypeError: Cannot read property 'get' of undefined` 
on this.